src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -986,102 +1341,143 @@ Actor* func_80A13400(EnGo* this, PlayState* play) {
+    f32 velocity = this->actor.velocity.y + this->actor.gravity;
+    s32 isFinished;
 
-    if (this->actor.bgCheckFlags & 2) {
-        ret = true;
+    if (this->actor.bgCheckFlags & 2) { // Settled to the Ground
+        // Transition 3->DONE
+        isFinished = true;
     } else {
-        if (temp_f14 > 0.0f) {
-            temp_f0 = temp_f14 / arg1;
-            this->actor.shape.rot.x += (s16)(9100.0f * temp_f0);
-            this->unk_3AE = 0;
+        if (velocity > 0.0f) {
+            // PHASE 1: Goron is rising
+            rotAndDistortFactor = velocity / initialVelocity;
+            this->actor.shape.rot.x += (s16)(0x238C * rotAndDistortFactor); // Max 50 degrees
+            this->elapsedHangtime = 0;
         } else if ((s32)this->actor.velocity.y == 0) {
-            if (arg3 >= this->unk_3AE) {
-                temp_f0 = (f32)this->unk_3AE / arg3;
+            if (maxHangtime >= this->elapsedHangtime) {
+                // PHASE 2: Goron is hanging in the air
+                rotAndDistortFactor = (f32)this->elapsedHangtime / maxHangtime;
             } else {
-                this->actor.gravity = -6.0f;
+                // Transition 2->3: Hanging is complete
+                this->actor.gravity = ENGO_POUND_FALL_VEL;
             }
-            this->unk_3AE++;
-        } else if (this->unk_3AE == 0) {
+            this->elapsedHangtime++;
+        } else if (this->elapsedHangtime == 0) {
+            // Transition 1->2: Goron has reached the peak
             this->actor.velocity.y = 0.0f;
             this->actor.gravity = 0.0f;
-            temp_f0 = temp_f14 / arg1;
+            rotAndDistortFactor = velocity / initialVelocity;
+        } else {
+            // PHASE 3: Goron is falling (Just letting gravity take effect)
         }
-        ret = false;
+        isFinished = false;
     }
 
-    this->unk_3A8 = (1.0f - temp_f0) * arg2;
-    this->actor.scale.x = this->unk_3A4 - this->unk_3A8;
-    this->actor.scale.y = this->unk_3A4 + this->unk_3A8;
+    // Handle Shape distortions
+    this->scaleFactorPoundDistortion = (1.0f - rotAndDistortFactor) * maxDistortion;
+    this->actor.scale.x = this->scaleFactor - this->scaleFactorPoundDistortion;
+    this->actor.scale.y = this->scaleFactor + this->scaleFactorPoundDistortion;
+    this->actor.scale.z = MAX(this->actor.scale.x, this->actor.scale.y);
 
-    if (this->actor.scale.y < this->actor.scale.x) {
-        this->actor.scale.z = this->actor.scale.x;
-    } else {
-        this->actor.scale.z = this->actor.scale.y;
-    }
-    return ret;
+    return isFinished;
 }
 
-void func_80A136B8(PlayState* play, s16 arg1, s16 arg2, s16 arg3) {
-    s16 sp26 = Quake_Add(Play_GetCamera(play, CAM_ID_MAIN), 3);
+/**
+ * Create the quake caused by the gatekeeper's pound.
+ */
+void EnGo_GatekeeperGoron_CreateQuake(PlayState* play, s16 quakeSpeed, s16 vertMagnitude, s16 quakeTime) {
+    s16 indexQuake = Quake_Add(Play_GetCamera(play, CAM_ID_MAIN), 3);
```

Comment:
```suggestion
void EnGo_GatekeeperGoron_QuakeAdd(PlayState* play, s16 speed, s16 verticalMag, s16 countdown) {
    s16 quakeIndex = Quake_Add(Play_GetCamera(play, CAM_ID_MAIN), 3);
```
Not a full review, but using these names will help sync up with https://github.com/zeldaret/mm/pull/1102